File: client/src/main/java/org/example/commands/InitialDataCommand.java
package org.example.commands;

import org.example.CommandType;
import org.example.data.ComputationalData;

public class InitialDataCommand implements Command {
    private ComputationalData computationalData;

    public InitialDataCommand(ComputationalData computationalData) {
        this.computationalData = computationalData;
    }

    @Override
    public CommandType getCommandType() {
        return CommandType.INITIAL_DATA;
    }

    public ComputationalData getComputationalData() {
        return computationalData;
    }

    public void setComputationalData(ComputationalData computationalData) {
        this.computationalData = computationalData;
    }
}

File: node-client/src/commands/InitialDataCommand.ts
import { CommandType } from '../CommandType';
import { Command } from './Command';
import { ComputationalData } from '../data/ComputationalData';

export class InitialDataCommand implements Command {
    public readonly commandType: CommandType = CommandType.INITIAL_DATA;

    constructor(public computationalData: ComputationalData) {}
}
